l5kit/l5kit/rasterization/box_rasterizer.py

```diff
@@ -55,14 +54,14 @@ def draw_boxes(
     else:
         im = np.zeros((raster_size[1], raster_size[0], 3), dtype=np.uint8)
 
-    box_world_coords = np.zeros((len(agents), 4, 2))
-    corners_base_coords = np.asarray([[-1, -1], [-1, 1], [1, 1], [1, -1]])
+    corners_base_coords = (np.asarray([[-1, -1], [-1, 1], [1, 1], [1, -1]]) / 2)[None, :, :]
 
     # compute the corner in world-space (start in origin, rotate and then translate)
-    for idx, agent in enumerate(agents):
-        corners = corners_base_coords * agent["extent"][:2] / 2  # corners in zero
-        r_m = yaw_as_rotation33(agent["yaw"])
-        box_world_coords[idx] = transform_points(corners, r_m) + agent["centroid"][:2]
+    corners_m = corners_base_coords * agents["extent"][:, None, :2]  # corners in zero
+    s = np.sin(agents["yaw"])
+    c = np.cos(agents["yaw"])
+    rotation_m = np.moveaxis(np.array(((c, -s), (s, c))), 2, 0)
```

Comment:
this part is also equivalent (tested) to:
```python
 rotation_m = np.moveaxis(np.array(((c, s), (-s, c))), 2, 0)
 box_world_coords = corners_m @ rotation_m + agents["centroid"][:, None, :2]
```
Which one do you think is the most clear in your opinion?

I'm not familiar with einsum so for me that part is magic, but at the same time my solution has a clockwise matrix to avoid transposing the points. I don't really know lol

Reply:
Good idea. I think matrix multiplication is still faster than einsum because there might be some optimization for it. I will change that.

Reply:
Speed is similar to the einsum. So I changed to using matrix multiplication for easier read.